spell/psm-checklists.md

```diff
@@ -0,0 +1,137 @@
+        - [ ] Deployer is no longer the `owner`
+        - [ ] `MCD_PAUSE_PROXY` is the `owner` of the contract (i.e. `require(MomLike(MOM).owner() == MCD_PAUSE_PROXY`))
+    - `LitePsmJob`
+        - [ ] Uses the default Solidity version in [`dss-cron`](https://github.com/makerdao/dss-cron)
+        - [ ] Optimizer is **enabled**
+        - [ ] Deployed contract matches the code in the [repo](https://github.com/makerdao/dss-cron)
+        - Constructor params:
+            - [ ] `_sequencer` matches `CRON_SEQUENCER` address from the chainlog
+            - [ ] `_litePsm` matches `DssLitePsm` address
+            - [ ] `_rushThreshold` matches the Exec Sheet (it might be named as "fill threshold")
+            - [ ] `_gushThreshold` matches the Exec Sheet (it might be named as "trim threshold")
+            - [ ] `_cutThreshold` matches the Exec Sheet (it might be named as "chug threshold")
+- Initialization
+    - `DssLitePsm`:
+        - [ ] `buf` is set to the value specified in the Exec Sheet
+        - [ ] `vow` is set to `MCD_VOW` from the chainlog
+        - [ ] `MCD_PAUSE_PROXY` is whitelisted to execute swaps with no fees (i.e. `KissLike(LITE_PSM).kiss(MCD_PAUSE_PROXY)`)
+        - [ ] `DssLitePsmMom` is authed (i.e. `RelyLike(LITE_PSM).rely(MOM)`)
+    - `DssLitePsmMom`:
+        - [ ] The chief (`MCD_ADM`) is set as the `authority` (i.e. `MomLike(MOM).setAuthority(MCD_ADM)`)
+    - `LitePsmJob`:
+        - [ ] Job is added to `CRON_SEQUENCER`
+    - [ ] `MCD_VAT`: new ilk is initialized
+    - [ ] `MCD_JUG`: new ilk is initialized
+    - `MCD_SPOT`:
+        - [ ] Sanity check: Dai parity (`par`) current value is 1 (`1 * RAY`)
+        - [ ] Collateralization ratio (`mat`) is set to 100% (`1 * WAD`) for the ilk 
+        - IF a new `pip` is being used:
+            - [ ] `pip` is set for the ilk
+            - [ ] `pip` is a `DSValue` instance
+            - [ ] The value on `pip` is set to 1 (`1 * WAD`)
+            - [ ] `pip` deployer is no longer the `owner`
+            - [ ] `MCD_PAUSE_PROXY` is the `owner` on `pip`
+        - OTHERWISE when reusing an existing `pip`:
+            - [ ] `pip` is set for the ilk
+            - [ ] `pip` in the chainlog matches `gem` symbol (i.e. `PIP_USDC` for `USDC`)
+    - [ ] `ILK_REGISTRY`: new ilk is added to the registry
+    - `CHAINLOG`:
+        - [ ] `DssLitePsm` is added to the chainlog and `addresses_mainnet.sol` according to the Exec Sheet
+        - [ ] `pocket` is added to the chainlog and `addresses_mainnet.sol` according to the Exec Sheet
+        - [ ] `DssLitePsmMom` is added to the chainlog and `addresses_mainnet.sol` according to the Exec Sheet
+        - [ ] `LitePsmJob` is added to the chainlog and `addresses_mainnet.sol` according to the Exec Sheet
+        - [ ] IF a new `pip` is being added, it is added to the chainlog and `addresses_mainnet.sol` as `PIP_{TOKEN_SYMBOL}` (i.e. `PIP_USDC`)
+- Test coverage:
```

Comment:
Can we please add end-to-end test here to check main functionally of the `DssLitePsm`?

As far as I can see, PSMs are not tested by any of the base tests, and `testPSMs` have to be specifically extended to call `_checkPsmIlkIntegration` for this each PSM.

Reply:
Good point.
`_checkPsmIlkIntegration` is not LitePSM friendly, it assumes there will be `clip` and a `gemJoin` for all PSMs.
Maybe a simple test to buy and sell gems with it would suffice. Not sure if it would belong in the base tests though.

Reply:
Since we expect to deploy more LitePSM modules in the future, I think it makes sense to add a base test to do most of the checks described in the `Test coverage`. Having a checklist is good, but having a base/reusable/configurable test is even better in my opinion.